### DISCUSSION TITLE
Is there a way to omit the as prop?

### DISCUSSION THREAD

**Discussion by d-henn** [sannajammeh/tw-classed#145] [General]
Just wondering if there's an easy way to gain the power of using the variants, but exclude the `as` prop. For example, I want to create a **Button** component, but I only ever want this to be rendered as a button and nothing else. At the moment someone could attempt to use the button as an anchor:

```typescript
<Button as="a" variant="filled">Button</Button>
 // renders <a ... />
```

**vs.**

```typescript
<Button as="a" variant="filled">Button</Button> //<-- Typescript will complain about the prop
// renders <button ... />
```

I know I can create my own component with a `forwardRef` and only include my props and force the button component, but wondering if there is an easy way to leverage the `deriveClassed` experience for this and still omit/constrain the `as` prop—both from a rendering level and typescript type level.

Thank you! :)

**Comment by sannajammeh** [comment]
Thanks for the question! Theres no easy way to do this as its fully intended behavior. ForwardRef would be the way to go.

This library intends to match the behaviour of stitches nearly 1-1. 

Personally, I think you should allow any underlying component to be rendered as most buttons in server rendered UI's are actually anchor tags.
